library_generation/owlbot/synthtool/languages/java.py

```diff
@@ -0,0 +1,691 @@
+ */
+"""
+PROTOBUF_HEADER = "// Generated by the protocol buffer compiler.  DO NOT EDIT!"
+BAD_LICENSE = """/\\*
+ \\* Copyright \\d{4} Google LLC
+ \\*
+ \\* Licensed under the Apache License, Version 2.0 \\(the "License"\\); you may not use this file except
+ \\* in compliance with the License. You may obtain a copy of the License at
+ \\*
+ \\* http://www.apache.org/licenses/LICENSE-2.0
+ \\*
+ \\* Unless required by applicable law or agreed to in writing, software distributed under the License
+ \\* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ \\* or implied. See the License for the specific language governing permissions and limitations under
+ \\* the License.
+ \\*/
+"""
+DEFAULT_MIN_SUPPORTED_JAVA_VERSION = 8
+METADATA = "metadata"
+LIBRARIES_BOM_VERSION = "libraries_bom_version"
+LIBRARIES_BOM_VERSION_ENV_KEY = "SYNTHTOOL_LIBRARIES_BOM_VERSION"
+HEADER_REGEX = re.compile("\\* Copyright \\d{4} Google LLC")
+LIBRARY_VERSION = "library_version"
+LIBRARY_VERSION_ENV_KEY = "SYNTHTOOL_LIBRARY_VERSION"
+
+
+def _file_has_header(path: Path) -> bool:
+    """Return true if the file already contains a license header."""
+    with open(path, "rt") as fp:
+        for line in fp:
+            if HEADER_REGEX.search(line):
+                return True
+    return False
+
+
+def _filter_no_header(paths: Iterable[Path]) -> Iterable[Path]:
+    """Return a subset of files that do not already have a header."""
+    for path in paths:
+        anchor = Path(path.anchor)
+        remainder = str(path.relative_to(path.anchor))
+        for file in anchor.glob(remainder):
+            if not _file_has_header(file):
+                yield file
+
+
+def fix_proto_headers(proto_root: Path) -> None:
+    """Helper to ensure that generated proto classes have appropriate license headers.
+
+    If the file does not already contain a license header, inject one at the top of the file.
+    Some resource name classes may contain malformed license headers. In those cases, replace
+    those with our standard license header.
+    """
+    s.replace(
+        _filter_no_header([proto_root / "src/**/*.java"]),
+        PROTOBUF_HEADER,
+        f"{GOOD_LICENSE}{PROTOBUF_HEADER}",
+    )
+    # https://github.com/googleapis/gapic-generator/issues/3074
+    s.replace(
+        [proto_root / "src/**/*Name.java", proto_root / "src/**/*Names.java"],
+        BAD_LICENSE,
+        GOOD_LICENSE,
+    )
+
+
+def fix_grpc_headers(grpc_root: Path, package_name: str = "unused") -> None:
+    """Helper to ensure that generated grpc stub classes have appropriate license headers.
+
+    If the file does not already contain a license header, inject one at the top of the file.
+    """
+    s.replace(
+        _filter_no_header([grpc_root / "src/**/*.java"]),
+        "^package (.*);",
+        f"{GOOD_LICENSE}package \\1;",
+    )
+
+
+def latest_maven_version(group_id: str, artifact_id: str) -> Optional[str]:
```

Comment:
I think we don't need this function once #3007 is merged. We can keep it for now.

Reply:
Now #3007 is merged, could you create a follow up PR to remove this function?